### PR TITLE
[Fixtures] Taxon factory

### PR DESCRIPTION
**Pull request by loic425**
| Q               | A
| --------------- | -----
| Branch?         | foundry-fixtures
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| Related tickets |
| License         | MIT

### CODE REVIEW

**Review by Roshyo — APPROVED** [review]

**Review by vvasiloi — CHANGES_REQUESTED** [review]
It should also have `withSlug`.

**Comment by vvasiloi** [comment]
I have more feedback, but it's about the logic that is being migrated, so it's better to address that later.
Let's first do the migration of current logic, then work on improving it.

**Review by vvasiloi — APPROVED** [review]

File: tests/DataFixtures/Factory/TaxonFactoryTest.php [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+        // testing fr_FR translation
+        $taxon->setCurrentLocale('fr_fr');
+        $taxon->setFallbackLocale('fr_FR');
+
+        $this->assertEquals('boardgames', $taxon->getSlug());
+    }
+
+    /** @test */
+    function it_creates_taxon_with_descriptions_for_each_locales(): void
```
Comment (vvasiloi) [inline tests/DataFixtures/Factory/TaxonFactoryTest.php:94]:
```suggestion
    function it_creates_taxon_with_description_for_each_locale(): void
```

File: tests/DataFixtures/Factory/TaxonFactoryTest.php [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+        // testing fr_FR translation
+        $taxon->setCurrentLocale('fr_fr');
+        $taxon->setFallbackLocale('fr_FR');
+
+        $this->assertEquals('Board games', $taxon->getName());
+        $this->assertEquals('board-games', $taxon->getSlug());
+    }
+
+    /** @test */
+    function it_creates_taxon_with_slugs_for_each_locales(): void
```
Comment (vvasiloi) [inline tests/DataFixtures/Factory/TaxonFactoryTest.php:71]:
```suggestion
    function it_creates_taxon_with_slug_for_each_locale(): void
```

File: tests/DataFixtures/Factory/TaxonFactoryTest.php [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+    /** @test */
+    function it_creates_taxon_with_given_code(): void
+    {
+        $taxon = TaxonFactory::new()->withCode('board-games')->create();
+
+        $this->assertEquals('board-games', $taxon->getCode());
+    }
+
+    /** @test */
+    function it_creates_taxon_with_names_for_each_locales(): void
```
Comment (vvasiloi) [inline tests/DataFixtures/Factory/TaxonFactoryTest.php:46]:
```suggestion
    function it_creates_taxon_with_name_for_each_locale(): void
```

**Review by vvasiloi — APPROVED** [review]

**Review by vvasiloi — APPROVED** [review]

**Pull request merged**